src/utils.ts

```diff
@@ -1,7 +1,7 @@
 import fetch from 'cross-fetch';
 import { Interface } from '@ethersproject/abi';
 import { Contract } from '@ethersproject/contracts';
-import { namehash } from '@ethersproject/hash';
+import namehash from '@ensdomains/eth-ens-namehash';
```

Comment:
Isnt this library making app bundle much bigger? Any way to check size before and after. Also maybe you could use `import { hash } from '@ensdomains/eth-ens-namehash'` instead

Reply:
@bonustrack this package already exists in [snapshot UI](https://github.com/snapshot-labs/snapshot/commit/6384fadc3f40674c37b4e4565237f23c1d9aea98), so there is no change in bundle size

![image](https://user-images.githubusercontent.com/15967809/154904882-59d6a573-b535-4d6a-9fc8-788e86e7f44b.png)


I will create a beta version for this, so you can check from the snapshot repo directly (if needed)

Reply:
There is not much difference in snapshot.js package

https://www.npmjs.com/package/@snapshot-labs/snapshot.js/v/0.3.49-beta2 - Unpacked Size - 1.33 MB
https://www.npmjs.com/package/@snapshot-labs/snapshot.js/v/0.3.48 - Unpacked Size - 1.14 MB

Reply:
You can also test/check the size on UI here:

https://github.com/snapshot-labs/snapshot/pull/1785
https://bafybeic6nkb6acveajxgnrikmw7qi6hl2t2krhwbmqmv76cjkmfwavkh7q.on.fleek.co/stats.html
https://snapshot.org/stats.html

Reply:
Ok then its fine, maybe we can try use destructured import here <https://github.com/snapshot-labs/snapshot/blob/35d28b8716d24eba8370b7ffcabf699b314c277d/src/components/SetupController.vue#L4> ENS dependency take a bit some space: <img width="634" alt="image" src="https://user-images.githubusercontent.com/16245250/154917112-f1c865ab-bfd4-4e28-810a-97a127323810.png">
Or maybe we can use the encoding ENS lib is using to support emoji and stay with Ethers.js which should be smaller

Reply:
> Or maybe we can use the encoding ENS lib is using to support emoji and stay with Ethers.js which should be smaller

ENS lib is using two packages to support emoji, https://github.com/ensdomains/eth-ens-namehash/blob/master/index.js,
If we use the same, we get the same size on snapshot.js too right 😅

Reply:
> Ok then it's fine, maybe we can try use destructured import here https://github.com/snapshot-labs/snapshot/blob/35d28b8716d24eba8370b7ffcabf699b314c277d/src/components/SetupController.vue#L4

Destructuring may not help reduce size because there are two functions on this package, [namehash is using normalize inside ](https://github.com/ensdomains/eth-ens-namehash/blob/master/index.js#L11) so it will same size

Reply:
Encoding shouldnt be that hard, we already made emoji domain works without libraries example: https://snapshot.org/#/%F0%9F%8D%AFdao.eth

Reply:
Maybe this is not a good example, it has direct IPFS settings but I know we have others with emoji and IPNS url and works fine

Reply:
> but I know we have others with emoji and IPNS url and works fine
This is because the spaces were created before when we were using this package instead of ethers.js, right now if there are spaces with emojis, owners won't be able to update the settings

I can't find any spaces with emojis other than 
https://snapshot.org/#/%F0%9F%8D%AFdao.eth/settings
https://snapshot.org/#/%F0%9F%A5%93%F0%9F%AA%99%F0%9F%97%B3.eth/settings

If you go to these spaces, we get this issue (there is a open issue on ethers.js https://github.com/ethers-io/ethers.js/issues/2376 )

If you go to other non-Ascii values spaces (other languages ) they work fine with ethers.js

like: https://snapshot.org/#/%F0%9F%8D%AFdao.eth